data_collection/gazette/spiders/sp/sp_suzano.py

```diff
@@ -0,0 +1,42 @@
+import datetime
+import re
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpSuzanoSpider(BaseGazetteSpider):
+    name = "sp_suzano"
+    TERRITORY_ID = "3552502"
+    start_date = datetime.date(2017, 8, 1)
+    start_urls = ["https://suzano.sp.gov.br/web/transparencia/imprensa-oficial/"]
+
+    def parse(self, response):
+        info = response.xpath("/html/body/div[2]/div/div/div[2]/div/div/*/*/span[2]")
+        data = response.xpath(
+            "/html/body/div[2]/div/div/div[2]/div/div/*/div[@class='desc']"
+        )
+        file_pattern = r'href="(https://[^"]+\.pdf)"'
+        date_pattern = r"(\d{1,2}[./]\d{1,2}[./]\d{4})"
+        for index, element in enumerate(data):
+            title = info[index].get()
+            matches_url = re.findall(file_pattern, element.get())
+            matches_date = re.findall(date_pattern, title)
+            if matches_url and matches_date:
+                url = matches_url[0]
```

Comment:
Aqui seria bom mudar algumas coisas:
1. Estamos usando `re.findall` para buscar a url, quando poderíamos usar um seletor pro atributo `href` (isso até faz com que possamos coletar arquivos não-PDF como .docx na edição 20.2 de 2021-01-28)
    1.1. Imagino que você já tenha detectado isso, mas agora seria necessário verificar se há ou não urls (casos onde não aparecem urls ou aparece apenas uma ao invés do comum que seria a url da imagem primeiro e a url do arquivo por último)
2. Outra coisa também é que estávamos usando o `re.findall` pra pegar apenas a primeira ocorrência. Nesse caso é melhor usar [`re.search`](https://docs.python.org/pt-br/3/library/re.html#re.search). Usar o método `.re_first` de uma resposta de seletor também é uma boa alternativa pois às vezes nem precisaríamos importar a biblioteca `re`.
3. Achei que valia adicionar um log de debug para linhas sem data
4. Ao invés de verificar `if matches_url and matches_date:` e continuar com o código dentro da verificação é interessante fazer esses curto-circuitos em Python pro código não expandir demais horizontalmente (nesse caso fazia sentido separar em 2 verificações distintas, mas também poderia verificar a negação `not matches_url or not matches_date`)
5. Apesar de o pipeline estar dando conta nesse caso, temos dado preferência a manter isso no código dos raspadores por padrão
```suggestion
            urls = url_box.css("a::attr(href)").getall()
            if not urls:
                continue
            url = urls[-1]

            date_raw = title_raw.re_first(r"\d{1,2}[./]\d{1,2}[./]\d{4}")
            if not date_raw:
                self.logger.debug(f"Não é possível encontrar a data do elemento com título '{title_raw.get()}'. URL de referência: {url}")
                continue

            if self.start_date > date > self.end_date:
                continue

```